Basic-Car-Maintenance/Shared/Dashboard/ViewModels/DashboardViewModel.swift

```diff
@@ -78,6 +78,21 @@ class DashboardViewModel: ObservableObject {
         }
     }
     
+    func updateEvent(_ maintenanceEvent: MaintenanceEvent) async {
+            
+            if let uid = authenticationViewModel.user?.uid {
+                guard let id = maintenanceEvent.id else {return}
+                var eventToUpdate = maintenanceEvent
+                eventToUpdate.userID = uid
+                try? Firestore
+                    .firestore()
+                    .collection("maintenance_events")
+                    .document(id)
+                    .setData(from: eventToUpdate)
+            }
+                    await self.getMaintenanceEvents()
+        }
+    
```

Comment:
The indentation is off for inside this function, and can you handle the error that comes back from Firebase if the `setData` doesn't work? An alert should be shown